Neo4j.Driver/Neo4j.Driver.Tests.Integration/Examples.cs

```diff
@@ -21,1127 +21,1077 @@
+    {
+        public ResultConsumeExample(ITestOutputHelper output, StandAloneIntegrationTestFixture fixture)
+            : base(output, fixture)
+        {
         }
 
-        public class ReadWriteTransactionExample : BaseExample
+        // tag::result-consume[]
+        public List<string> GetPeople()
         {
-            public ReadWriteTransactionExample(ITestOutputHelper output, StandAloneIntegrationTestFixture fixture)
-                : base(output, fixture)
+            using var session = Driver.Session();
+            return session.ExecuteRead(tx =>
             {
-            }
+                var result = tx.Run("MATCH (a:Person) RETURN a.name ORDER BY a.name");
+                return result.Select(record => record[0].As<string>()).ToList();
+            });
+        }
+        // end::result-consume[]
 
-            [RequireServerFact]
-            public void TestReadWriteTransactionExample()
-            {
-                // When & Then
-                AddPerson("Alice").Should().BeGreaterOrEqualTo(0L);
-            }
+        [RequireServerFact]
+        public void TestResultConsumeExample()
+        {
+            // Given
+            Write("CREATE (a:Person {name: 'Alice'})");
+            Write("CREATE (a:Person {name: 'Bob'})");
+            // When & Then
+            GetPeople().Should().Contain(new[] {"Alice", "Bob"});
+        }
+    }
 
-            // tag::read-write-transaction[]
-            public long AddPerson(string name)
-            {
-                using (var session = Driver.Session())
-                {
-                    session.WriteTransaction(tx => CreatePersonNode(tx, name));
-                    return session.ReadTransaction(tx => MatchPersonNode(tx, name));
-                }
-            }
+    public class ResultRetainExample : BaseExample
+    {
+        public ResultRetainExample(ITestOutputHelper output, StandAloneIntegrationTestFixture fixture)
+            : base(output, fixture)
+        {
+        }
 
-            private static IResultSummary CreatePersonNode(ITransaction tx, string name)
-            {
-                return tx.Run("CREATE (a:Person {name: $name})", new {name}).Consume();
-            }
+        // tag::result-retain[]
+        public int AddEmployees(string companyName)
+        {
+            using var session = Driver.Session();
+            var persons = session.ExecuteRead(tx => tx.Run("MATCH (a:Person) RETURN a.name AS name").ToList());
+            return persons.Sum(person => session.ExecuteWrite(tx =>
+            {
+                var result = tx.Run("MATCH (emp:Person {name: $person_name}) " +
+                                    "MERGE (com:Company {name: $company_name}) " +
+                                    "MERGE (emp)-[:WORKS_FOR]->(com)",
+                    new {person_name = person["name"].As<string>(), company_name = companyName});
+                result.Consume();
+                return 1;
```

Comment:
probably just being dumb but why does this just return a hardcoded 1?

Reply:
I don't really know, it was in the example before.

Reply:
Haha. Thanks for the answer!